src/post/dto/create-post.dto.ts

```diff
@@ -1,11 +1,23 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { ActiveStatus } from 'src/constants';
+import { IsNotEmpty, IsOptional, IsNumber, IsString, MaxLength } from 'class-validator';
+import { Transform } from 'class-transformer';
 export class CreatePostDto {
-  readonly userId;
+  @ApiProperty()
+  @IsNotEmpty({ message: 'User id must not be empty' })
+  @IsNumber({ allowNaN: false, allowInfinity: false }, { message: 'User id must be a number' })
+  readonly userId: number;
 
-  readonly content;
+  @ApiProperty()
+  @Transform(({ value }) => value?.trim())
```

Comment:
I think the order of validator annotations is not correct. The execution order of the validator will be from bottom to top.
Suggestion:
`@MaxLength(10000, { message: 'Content must be less than 10000 characters' })
  @IsNotEmpty({ message: 'Content must not be empty' })
  @Transform(({ value }) => value?.trim())
  @IsString({ message: 'Content must be a string' })
  readonly content: string;`